### DISCUSSION TITLE
How to get the mangled symbol for a FunctionSymbol

### DISCUSSION THREAD

**Discussion by matcool** [NationalSecurityAgency/ghidra#3517] [Q&A]
I have a FunctionSymbol of a certain function that has a symbol, and i cannot figure out a way to get its mangled symbol which is very easily shown on the ghidra interface
![image](https://user-images.githubusercontent.com/26722564/137647408-711ef87c-c2fe-424b-a782-bf1eaf584b40.png)

**Comment by astrelsky** [comment]
Try getting all the symbols at the entry point address of the function. The mangled symbol there is not the primary symbol so you will need to get all of them. **If** there isn't a method in the `FlatProgramAPI` for this try the `SymbolTable` obtained from `currentProgram.getSymbolTable()`.
